packages/shorebird_cli/lib/src/commands/releases/delete_releases_command.dart

```diff
@@ -0,0 +1,96 @@
+import 'dart:async';
+
+import 'package:collection/collection.dart';
+import 'package:mason_logger/mason_logger.dart';
+import 'package:shorebird_cli/src/auth_logger_mixin.dart';
+import 'package:shorebird_cli/src/command.dart';
+import 'package:shorebird_cli/src/shorebird_config_mixin.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+
+/// {@template delete_releases_command}
+///
+/// `shorebird releases delete`
+/// Delete the specified release.
+/// {@endtemplate}
+class DeleteReleasesCommand extends ShorebirdCommand
+    with AuthLoggerMixin, ShorebirdConfigMixin {
+  /// {@macro delete_releases_command}
+  DeleteReleasesCommand({
+    required super.logger,
+    super.auth,
+    super.buildCodePushClient,
+  }) {
+    argParser.addOption(
+      'version',
+      help: 'The release version to delete.',
+    );
+  }
+
+  @override
+  String get name => 'delete';
+
+  @override
+  String get description => 'Delete the specified release version.';
+
+  @override
+  Future<int> run() async {
+    if (!auth.isAuthenticated) {
+      printNeedsAuthInstructions();
+      return ExitCode.noUser.code;
+    }
+
+    if (!hasShorebirdYaml) {
+      logger.err(
+        '''Shorebird is not initialized. Did you run ${lightCyan.wrap('shorebird init')}?''',
+      );
+      return ExitCode.config.code;
+    }
+
+    final appId = getShorebirdYaml()!.appId.value!;
+
+    final codePushClient = buildCodePushClient(
+      httpClient: auth.client,
+      hostedUri: hostedUri,
+    );
+
+    final List<Release> releases;
+    try {
+      releases = await codePushClient.getReleases(appId: appId);
+    } catch (error) {
+      logger.err('$error');
+      return ExitCode.software.code;
+    }
+
+    final versionInput = results['version'] as String? ??
+        logger.prompt(
+          '${lightGreen.wrap('?')} Which version would you like to delete?',
+        );
+
+    final releaseToDelete = releases.firstWhereOrNull(
+      (release) => release.version == versionInput,
+    );
+    if (releaseToDelete == null) {
+      logger.err('No release found for version "$versionInput"');
+      return ExitCode.software.code;
+    }
+
+    final shouldDelete = logger.confirm(
+      'Are you sure you want to delete release ${releaseToDelete.version}?',
+    );
+    if (!shouldDelete) {
+      logger.info('Aborted.');
+      return ExitCode.success.code;
+    }
+
```

Comment:
should we show some progress here to give the user feedback that we're deleting the release in case it takes a while?

Reply:
Good call, done.